File: go/internal/shared/digits.go
// Package shared holds small helpers reused by more than one validator.
// Kept internal: these are implementation details, not part of the public
// API of any validator package.
package shared

import "regexp"

var nonDigits = regexp.MustCompile(`\D`)

// RemoveNonDigits strips every character that is not 0-9.
func RemoveNonDigits(value string) string {
	return nonDigits.ReplaceAllString(value, "")
}

File: php/src/Shared/Digits.php
<?php

declare(strict_types=1);

namespace BrValidator\Shared;

/**
 * Small helpers reused by more than one validator. Not part of the public
 * API of any validator class.
 */
final class Digits
{
    private function __construct()
    {
    }

    public static function removeNonDigits(string $value): string
    {
        return preg_replace('/\D/', '', $value) ?? '';
    }
}
